### PR TITLE
Don't show builder after dot

### PR DESCRIPTION
**Pull request by rchande**
Tag @dotnet/roslyn-ide for review

### CODE REVIEW

File: src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs [hunk 1]
```diff
@@ -156,7 +156,21 @@ private bool IsLambdaExpression(SemanticModel semanticModel, int position, Synta
             // open the builder if any overload takes a delegate at our argument position
             var inferredTypeInfo = typeInferrer.GetTypeInferenceInfo(semanticModel, position, cancellationToken: cancellationToken);
 
-            return inferredTypeInfo.Any(type => GetDelegateType(type, semanticModel.Compilation).IsDelegateType());
+            if (inferredTypeInfo.Any(type => GetDelegateType(type, semanticModel.Compilation).IsDelegateType()))
+            {
+                // In the following situation, the type inferrer will infer Task to support target type preselection
+                // Action a = Task.$$
+                // We need to explicitly exclude invocation/member access from suggestion mode
+                var previousToken = token.GetPreviousTokenIfTouchingWord(position);
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs]: I'd much rather have the syntax check up front.  Asking if we're in a locatoin where you could have an inferred lambda.

**Comment by rchande** [comment]
Fixes https://github.com/dotnet/roslyn/issues/15443

File: src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs (hunk 1 above)
Comment (rchande) [inline src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs]: Done, thanks. This actually looks way better.

**Comment by rchande** [comment]
retest windows_debug_unit32_prtest please

File: src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs [hunk 2]
```diff
@@ -152,10 +152,19 @@ private bool IsLambdaExpression(SemanticModel semanticModel, int position, Synta
                 position = token.Parent.SpanStart;
             }
 
+            // In the following situation, the type inferrer will infer Task to support target type preselection
+            // Action a = Task.$$
+            // We need to explicitly exclude invocation/member access from suggestion mode
+            var previousToken = token.GetPreviousTokenIfTouchingWord(position);
+            if (previousToken.IsKind(SyntaxKind.DotToken) &&
+                previousToken.Parent.IsKind(SyntaxKind.InvocationExpression, SyntaxKind.SimpleMemberAccessExpression))
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs]: dot token can't be parented by an invocation expression.

**Pull request merged**